pkg/ebpf/netlink/bpf_sampler.go

```diff
@@ -0,0 +1,31 @@
+package netlink
+
+import (
+	"errors"
+	"math"
+
+	"golang.org/x/net/bpf"
+)
+
+var errInvalidSamplingRate = errors.New("sampling rate must be within (0, 1)")
+
+// GenerateBPFSampler returns BPF assembly for a traffic sampler
+func GenerateBPFSampler(samplingRate float64) ([]bpf.RawInstruction, error) {
```

Comment:
Is this unused at the moment? I couldn't find usages

Reply:
Nevermind! Found it in `consumer.go` - not sure why that didn't show up for me initially.